### PR TITLE
Update CI to use Go 1.18beta.2

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

**Review by benbp — APPROVED** [review]

File: sdk/azcore/internal/pipeline/request.go [hunk 1]
```diff
@@ -157,8 +157,7 @@ func (req *Request) Close() error {
 
 // Clone returns a deep copy of the request with its context changed to ctx.
 func (req *Request) Clone(ctx context.Context) *Request {
-	r2 := Request{}
-	r2 = *req
+	r2 := *req
```
Comment (richardpark-msft) [inline sdk/azcore/internal/pipeline/request.go:160]: Just checking - is this just a lint fix or is there something 1.18-interesting in here?

**Review by richardpark-msft — APPROVED** [review]
Very exciting!

File: sdk/azcore/internal/pipeline/request.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/azcore/internal/pipeline/request.go:160]: It's linter clean-up found in the latest version of the linter.

**Review by chlowell — APPROVED** [review]

**Pull request merged**